tests/parsenum/main.c

```diff
@@ -0,0 +1,97 @@
+#include <stdint.h>
+#include <stdio.h>
+
+#include "parsenum.h"
+#include "warnp.h"
+
+#define PARSENUM_CMP(x, y)						\
+	((isnan((double)x) && isnan((double)y)) ||			\
+	    (isinf((double)x) == isinf((double)y) && isinf((double)x)) || \
```

Comment:
(that's why I thought that `isinf() == isinf()` was more useful than it actually is)